### PR TITLE
Fix, update and enable achievement updates

### PR DESCRIPTION
**Pull request by sderickson**
* Clients check updated achievements as well as new ones
* Clients do not wait to keep checking
* Update achievement points along with everything else in EarnedAchievement.upsertFor
* Fix various bugs

### CODE REVIEW

File: app/core/application.coffee [hunk 1]
```diff
@@ -62,7 +62,24 @@ Application = {
     $(document).bind 'keydown', preventBackspace
     preload(COMMON_FILES)
     CocoModel.pollAchievements()
-#    @checkForNewAchievement() # TODO: Enable once thoroughly tested
+    unless me.get('anonymous')
+      # TODO: Remove logging later, once this system has proved stable
+      me.on 'change:earned', (user, newEarned) ->
+        oldEarned = user.previous('earned')
+        if oldEarned.gems isnt newEarned.gems
```
Comment (nwinter) [inline app/core/application.coffee]: `oldEarned` might be `undefined`

File: app/schemas/models/user.coffee [hunk 2]
```diff
@@ -344,7 +344,7 @@ _.extend UserSchema.properties,
   schoolName: {type: 'string'}
   role: {type: 'string', enum: ["God", "advisor", "parent", "principal", "student", "superintendent", "teacher", "technology coordinator"]}
   birthday: c.stringDate({title: "Birthday"})
-  lastAchievementChecked: c.objectId({ name: 'Last Achievement Checked' })
+  lastAchievementChecked: c.stringDate({ name: 'Last Achievement Checked' })
```
Comment (nwinter) [inline app/schemas/models/user.coffee]: Will this require any migration?

File: server/middleware/achievements.coffee [hunk 3]
```diff
@@ -19,7 +19,12 @@ module.exports =
     unless hasPermission or database.isJustFillingTranslations(req, achievement)
       throw new errors.Forbidden('Must be an admin, artisan or submitting translations to edit an achievement')
 
+    propsWatching = ['query', 'proportionalTo', 'rewards', 'worth']
```
Comment (nwinter) [inline server/middleware/achievements.coffee]: Also `function`

**Comment by nwinter** [comment]
Looks good apart from my notes.

File: app/schemas/models/user.coffee (hunk 2 above)
Comment (sderickson) [inline app/schemas/models/user.coffee]: No, it has been disabled and left unset on production, so unless someone manually found the endpoint and hit it during the last week, no need.

File: app/core/application.coffee (hunk 1 above)
Comment (sderickson) [inline app/core/application.coffee]: Fixed!

**Pull request merged**